### PR TITLE
Add link to whole search result tile

### PR DESCRIPTION
**Pull request by bazelajoanna**
Make the wole SearchResultTile clickable and able to navigate to ProblemDetailedPage.

### CODE REVIEW

File: cypress/integration/problem_detailed_page.js [hunk 1]
```diff
@@ -19,4 +19,10 @@ describe("Problem Detailed Page", () => {
     cy.get(`a[href="/problems/1"]`).first().click();
     cy.contains("Pobierz treść");
   });
+
+  it("shows problem detailed page after clicking on search result tile on mobile and tablet", () => {
```
Comment (evulins) [inline cypress/integration/problem_detailed_page.js]: I am just wondering if this test should be here or in the new file for problems page? What do you think?

File: src/components/molecules/SearchResultTile/index.tsx [hunk 2]
```diff
@@ -19,9 +18,9 @@ const SearchResultTile: FC<ProblemSummaryProps> = ({
   return (
     <S.SearchResultTileWrapper>
       <S.DetailsWrapper>
-        <Link to={`/problems/${id}`}>
-          <S.Title>{title}</S.Title>
-        </Link>
+        <S.Title>
+          <S.StyledTitleLink to={`/problems/${id}`}>{title}</S.StyledTitleLink>
+        </S.Title>
```
Comment (evulins) [inline src/components/molecules/SearchResultTile/index.tsx:23]: 👏

**Review by evulins — COMMENTED** [review]
Works like a charm! :)

File: cypress/integration/problem_detailed_page.js (hunk 1 above)
Comment (franiakat) [reply cypress/integration/problem_detailed_page.js]: There.

File: src/App.test.tsx [hunk 3]
```diff
@@ -85,4 +85,14 @@ describe("<App />", () => {
...
+
+  it("doesn't show problem detailed page after clicking on search result tile not clickable item on large dekstop", async () => {
+    testRenderer(<App />);
+    const author = (await screen.findByText(
+      "Maria Konopnicka"
+    )) as HTMLInputElement;
+    userEvent.click(author);
+
+    expect(await screen.findByText(/searchResults/)).toBeInTheDocument();
+  });
```
Comment (franiakat) [inline src/App.test.tsx]: I'd test it in cypress, specifying exactly the tool : >

File: cypress/integration/problem_detailed_page.js (hunk 1 above)
Comment (bazelajoanna) [reply cypress/integration/problem_detailed_page.js]: Right! 🐈

File: src/App.test.tsx (hunk 3 above)
Comment (bazelajoanna) [reply src/App.test.tsx]: I've moved it to cypress, by default it tests on screen with dimension 1000x600, so I haven't specified the tool.

**Review by franiakat — APPROVED** [review]
👏 nice Asia, I like the way it's implemented, thanks for the cypress tests as well!

**Pull request merged**